### ISSUE TITLE
Allow `tf.layers` naming conventions 

### ISSUE THREAD

**Issue by ppwwyyxx** [ppwwyyxx/tensorpack#627]
It's not wise to keep using a different naming conventions from tf.layers despite the historical reasons. 
Both argument names and variable names should be made consistent. And we can further consider layer names.

For argument names: can just put a conversion inside each function call to translate the old names to tf.layer names, so that both can be used.
For variable names: can add an option for each layer (or maybe globally) to choose which convention to use.
Follow #291.

Argument Names:
- [x] Allow the equivalent argument names in `tf.layers` convention.
- [x] Move examples to `tf.layers` convention.
- [x] BatchNorm not done yet.
- [ ] Support more options from `tf.layers` that aren't available in tensorpack.
